web/src/features/hfiCalculator/components/PrepLevelCell.tsx

```diff
@@ -84,11 +87,17 @@ const useStyles = makeStyles({
 const PrepLevelCell = (props: PrepLevelCellProps) => {
   const classes = useStyles()
 
-  const prepLevel = calculatePrepLevel(props.meanIntensityGroup)
+  const prepLevel = () => {
+    if (props.meanPrepLevel) {
+      return calculateMeanPrepLevel(props.meanIntensityGroup)
+    } else {
+      return calculatePrepLevel(props.meanIntensityGroup)
+    }
+  }
 
   const formatPrepLevelByValue = () => {
     if (!props.meanPrepLevel) {
```

Comment:
There's a lot of nested checking of `if (!props.meanPrepLevel)` - I wonder if this needs to be reconsidered? Should we not rather break out into two different variables?